src/helics/core/mpi/MpiComms.h

```diff
@@ -10,29 +10,96 @@ This software was co-developed by Pacific Northwest National Laboratory, operate
 #define _HELICS_MPI_COMMS_
 #pragma once
 
-#include "helics/core/CommsInterface.h"
+#include "../CommsInterface.hpp"
 #include <atomic>
+#include <set>
+#include <string>
+#include <future>
+#include <vector>
+#include <mutex>
+#include "helics/helics-config.h"
 
+#include <mpi.h>
+
+#if (BOOST_VERSION_LEVEL >=2)
+namespace boost
+{
+    namespace asio
+    {
+        class io_context;
+        using io_service = io_context;
+    }
+}
+#else
+namespace boost
+{
+    namespace asio
+    {
+        class io_service;
+    }
+}
+#endif
```

Comment:
As far as I can tell you are not using the boost io service stuff so we can probably remove these definitions.